aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -0,0 +1,218 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws.glue;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.BaseMetastoreTableOperations;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.hadoop.HadoopFileIO;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.glue.GlueClient;
+import software.amazon.awssdk.services.glue.model.ConcurrentModificationException;
+import software.amazon.awssdk.services.glue.model.CreateTableRequest;
+import software.amazon.awssdk.services.glue.model.EntityNotFoundException;
+import software.amazon.awssdk.services.glue.model.GetTableRequest;
+import software.amazon.awssdk.services.glue.model.GetTableResponse;
+import software.amazon.awssdk.services.glue.model.Table;
+import software.amazon.awssdk.services.glue.model.TableInput;
+import software.amazon.awssdk.services.glue.model.UpdateTableRequest;
+
+public class GlueTableOperations extends BaseMetastoreTableOperations {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GlueTableOperations.class);
+
+  // same as org.apache.hadoop.hive.metastore.TableType.EXTERNAL_TABLE
+  // more details: https://docs.aws.amazon.com/glue/latest/webapi/API_TableInput.html
+  private static final String GLUE_EXTERNAL_TABLE_TYPE = "EXTERNAL_TABLE";
+
+  private final Configuration conf;
+  private final GlueClient glue;
+  private final String catalogId;
+  private final String databaseName;
+  private final String tableName;
+  private final String fullName;
+
+  private FileIO fileIO;
+
+  /**
+   * constructor
+   * @param conf Hadoop config, passed in for users of HadoopIO
+   * @param glue Glue client
+   * @param catalogId Glue catalog ID, which is the AWS account ID. When null, it uses the account of the Glue client.
+   * @param databaseName Glue database name
+   * @param tableName Glue table name
+   */
+  public GlueTableOperations(
+      Configuration conf,
+      GlueClient glue,
+      String catalogId,
+      String databaseName,
+      String tableName) {
+    this.conf = conf;
+    this.glue = glue;
+    this.catalogId = catalogId;
+    this.databaseName = databaseName;
+    this.tableName = tableName;
+    this.fullName = String.format("glue.%s.%s.%s", catalogId, databaseName, tableName);
+  }
+
+  @Override
+  public FileIO io() {
+    if (fileIO == null) {
+      fileIO = new HadoopFileIO(conf);
+    }
+    return fileIO;
+  }
+
+  /**
+   * glue.catalogId.databaseName.tableName
```

Comment:
This could just be a one liner

/**
* Returns table name in the form glue.catalogId.databaseName.tableName
**/